### PR TITLE
Use `-O4` when compiling benchmarks.

### PR DESCRIPTION
**Pull request by LorenVS**
`--trust-type-annotations` is deprecated.

### CODE REVIEW

File: protobuf/benchmarks/compile-js.sh [hunk 1]
```diff
@@ -10,7 +10,7 @@ OUTPUT_DIR="${SCRIPT_DIR}/temp"
 
 mkdir -p "${OUTPUT_DIR}"
 
-dart2js --trust-type-annotations        \
+dart2js --omit-implicit-checks          \
         --trust-primitives              \
```
Comment (rakudrama) [inline protobuf/benchmarks/compile-js.sh:14]: Combine these lines into `-O4`.

Comment (LorenVS) [reply protobuf/benchmarks/compile-js.sh]: Done

**Review by rakudrama — APPROVED** [review]

File: protobuf/benchmarks/compile-js.sh [hunk 2]
```diff
@@ -10,7 +10,6 @@ OUTPUT_DIR="${SCRIPT_DIR}/temp"
 
 mkdir -p "${OUTPUT_DIR}"
 
-dart2js --trust-type-annotations        \
-        --trust-primitives              \
+dart2js -O4                             \
```
Comment (osa1) [inline protobuf/benchmarks/compile-js.sh:13]: We should update this to use `dart compile js`, [dart2js documentation](https://dart.dev/tools/dart2js) says `dart2js` is deprecated:

> Version note: The dart2js command-line tool is deprecated as of Dart 2.17 and will be removed in a future release.
>
> Use [dart compile js](https://dart.dev/tools/dart-compile#js) or the [webdev](https://dart.dev/tools/webdev) tool to compile Dart code to deployable JavaScript.

File: protobuf/benchmarks/compile-js.sh [hunk 3]
```diff
@@ -10,7 +10,6 @@ OUTPUT_DIR="${SCRIPT_DIR}/temp"
 
 mkdir -p "${OUTPUT_DIR}"
 
-dart2js --trust-type-annotations        \
-        --trust-primitives              \
```
Comment (osa1) [inline protobuf/benchmarks/compile-js.sh:14]: So `-O4` enables `--trust-primitives`? I can't see this mentioned in [dart2js documentation](https://dart.dev/tools/dart2js).

**Review by osa1 — APPROVED** [review]
I'm not sure how common `-O4` is when targeting JS, but LGTM assuming that's what we want to optimize for.

File: protobuf/benchmarks/compile-js.sh (hunk 3 above)
Comment (osa1) [reply protobuf/benchmarks/compile-js.sh:14]: https://github.com/dart-lang/sdk/blob/40ad7aee40a82aba55a6a9481b9421c6785c9a2b/pkg/compiler/lib/src/options.dart#L787-L789

**Pull request merged**